GenericPester.Tests.ps1

```diff
@@ -0,0 +1,94 @@
+<#
+.SYNOPSIS
+    Test file for Pester that will ensure all non-test scripts are syntactically correct, meet best practices and include detailed comment-based help
+.PARAMETER Path
+    Optional. A list of folder or file paths that will be examined for scripts to test.  Defaults to the invocation path to support running via Invoke-Pester
+#>
+[CmdletBinding()]
+Param(
+    [Parameter(Mandatory = $false)]
+    [string[]]$Path = (Split-Path -Parent $MyInvocation.MyCommand.Path)
+)
+$scripts = Get-ChildItem $Path -Recurse -File -Include '*.ps1' -Exclude '*.Tests.ps1'
+
+foreach ($script in $scripts)
+{
+    $scriptPath = $script.FullName
+    Describe "$scriptPath" {    
+        $astErrors = @()
+        # Parse the function using AST
+        $ast = [System.Management.Automation.Language.Parser]::ParseFile($scriptPath, [ref]$null, [ref]$astErrors)
+
+        Context 'Structure and Syntax' {
+            It 'Contains no errors' { $astErrors.Count | Should BeLessThan 1 }
+            It 'Has a valid signiture (if signed)' {
+                $result = Get-AuthenticodeSignature $scriptPath
+                if ($result.Status -eq [System.Management.Automation.SignatureStatus]::NotSigned) {
```

Comment:
Hmmm, from download.squaredup.com sure but they shouldn't be signed in-repo (contributors shouldn't be required to have a trusted code signing cert, never mind ours).